hive/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -155,14 +164,155 @@ public void renameTable(TableIdentifier from, TableIdentifier to) {
       throw new org.apache.iceberg.exceptions.AlreadyExistsException("Table already exists: %s", to);
 
     } catch (TException e) {
-      throw new RuntimeException("Failed to rename " + from.toString() + " to " + to.toString(), e);
+      throw new RuntimeException("Failed to rename " + from + " to " + to, e);
 
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
     }
   }
 
+  @Override
+  public void createNamespace(Namespace namespace, ImmutableMap<String, String> meta) {
+    Preconditions.checkArgument(
+        !namespace.isEmpty(),
+        "Cannot create namespace with invalid name: %s", namespace);
+    Preconditions.checkArgument(
+        namespace.levels().length == 1,
+        "Cannot support multi part namespace in Hive MetaStore: %s", namespace);
+
+    try {
+      clients.run(client -> {
+        client.createDatabase(convertToDatabase(namespace, meta));
+        return null;
+      });
+
+    } catch (AlreadyExistsException e) {
+      throw new org.apache.iceberg.exceptions.AlreadyExistsException(e, "Namespace '%s' already exists!",
+            namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to create namespace " + namespace + " in Hive MataStore", e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(
+          "Interrupted in call to createDatabase(name) " + namespace + " in Hive MataStore", e);
+    }
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    if (namespace.levels().length > 1) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+    }
+
+    try {
+      return namespace.isEmpty() ? clients.run(
+          HiveMetaStoreClient::getAllDatabases)
+          .stream()
+          .map(Namespace::of)
+          .collect(Collectors.toList()) : Stream.of(Namespace.empty())
+          .collect(Collectors.toList());
```

Comment:
It doesn't make sense to combine these cases into a single expression. The case where the namespace is non-empty (not listing all databases) always produces the same result so it doesn't need to use `Stream` at all. It also doesn't need to be in the `try` block. It should look like this:

```java
  if (!namespace.isEmpty) {
    // Hive doesn't support nesting, so there are no namespaces
    return ImmutableList.of();
  }

  try {
    // list and return all databases
    return clients.run(HiveMetastoreClient::getAllDatabases)
        .stream()
        .map(Namespace.of)
        .collect(Collectors.toList());
  } catch (...) {
    ...
  }
```